mne/viz/epochs.py

```diff
@@ -891,44 +927,100 @@ def plot_epochs_psd(epochs, fmin=0, fmax=np.inf, tmin=None, tmax=None,
     """
+    from matplotlib.ticker import ScalarFormatter
     from .raw import _set_psd_plot_params, _convert_psds
     fig, picks_list, titles_list, units_list, scalings_list, ax_list, \
         make_label = _set_psd_plot_params(
             epochs.info, proj, picks, ax, area_mode)
+    if spatial_colors is None:
+        spatial_colors = False if average else True
 
+    psd_list = list()
+    ylabels = list()
     for ii, (picks, title, ax) in enumerate(zip(picks_list, titles_list,
                                                 ax_list)):
-        psds, freqs = psd_multitaper(epochs, picks=picks, fmin=fmin,
-                                     fmax=fmax, tmin=tmin, tmax=tmax,
-                                     bandwidth=bandwidth, adaptive=adaptive,
-                                     low_bias=low_bias,
-                                     normalization=normalization, proj=proj,
-                                     n_jobs=n_jobs)
+        if method == 'multitaper':
+            psds, freqs = psd_multitaper(epochs, picks=picks, fmin=fmin,
+                                         fmax=fmax, tmin=tmin, tmax=tmax,
+                                         bandwidth=bandwidth,
+                                         adaptive=adaptive,
+                                         low_bias=low_bias,
+                                         normalization=normalization,
+                                         proj=proj,
+                                         n_jobs=n_jobs)
+        elif method == 'welch':
+            psds, freqs = psd_welch(epochs, picks=picks, fmin=fmin,
+                                    fmax=fmax, tmin=tmin, tmax=tmax,
+                                    n_fft=256, n_overlap=0, n_per_seg=None,
+                                    proj=proj,
+                                    n_jobs=n_jobs)
+        else:
+            raise ValueError('Only "multitaper" and "welch" are supported.')
 
         ylabel = _convert_psds(psds, dB, 'auto', scalings_list[ii],
                                units_list[ii],
                                [epochs.ch_names[pi] for pi in picks])
 
         # mean across epochs and channels
-        psd_mean = np.mean(psds, axis=0).mean(axis=0)
-        if area_mode == 'std':
-            # std across channels
-            psd_std = np.std(np.mean(psds, axis=0), axis=0)
-            hyp_limits = (psd_mean - psd_std, psd_mean + psd_std)
-        elif area_mode == 'range':
-            hyp_limits = (np.min(np.mean(psds, axis=0), axis=0),
-                          np.max(np.mean(psds, axis=0), axis=0))
-        else:  # area_mode is None
-            hyp_limits = None
-
-        ax.plot(freqs, psd_mean, color=color)
-        if hyp_limits is not None:
-            ax.fill_between(freqs, hyp_limits[0], y2=hyp_limits[1],
-                            color=color, alpha=area_alpha)
+        psd_mean = np.mean(psds, axis=0)
+        if average:
```

Comment:
This snipped looks a hell of a lot like this one:
https://github.com/mne-tools/mne-python/blob/d7c8e865fbfe2dc8854956a446005b05a82300c6/mne/viz/raw.py#L779-L795

We could make a function in `viz/utils.py` and fully cover it. Which is not the case now.